### PR TITLE
Report all agent errors using the chat or status bar

### PR DESCRIPTION
**Pull request by pkukielka**
Fixes: Invalid access token reported
[#289](https://github.com/sourcegraph/jetbrains/issues/289)
[#762](https://github.com/sourcegraph/jetbrains/issues/762)

Now every chat or agent error is properly reported on the UI:

image

## Test plan

1. Create new access token
2. Login using that access token
3. Run some Cody actions, like chat question and autocomplete
4. Remove access token from SG instance
5. Run some Cody action and check if it properly reports an error
6. Switch account and check if error is gone
7. Switch account back and check if error is reported again

### CODE REVIEW

File: src/main/kotlin/com/sourcegraph/cody/statusbar/CodyAutocompleteStatus.kt [hunk 1]
```diff
@@ -34,6 +35,12 @@ enum class CodyAutocompleteStatus : PresentableEnum, WithIcon {
 
     override val icon: Icon = Icons.StatusBar.CodyAutocompleteUnavailable
   },
+  AgentError {
+    override fun getPresentableText(): String =
+        "Cody encountered an error: ${CodyAgentService.agentError.get()}"
+
+    override val icon: Icon = Icons.StatusBar.CodyAutocompleteUnavailable
```
Comment (mkondratek) [inline src/main/kotlin/com/sourcegraph/cody/statusbar/CodyAutocompleteStatus.kt]: nit: `s/CodyAutocompleteUnavailable/CodyUnavailable`?

File: src/main/resources/CodyBundle.properties [hunk 2]
```diff
@@ -43,6 +43,11 @@ chat.rate-limit-error.explain=\
     To ensure that Cody can stay operational for all Cody users, please come back tomorrow for more chats, commands, and autocompletes. \
     <a href=\"https://sourcegraph.com/docs/cody/core-concepts/cody-gateway#rate-limits-and-quotas\">Learn more.</a>\
   </html>
+chat.general-error=\
+  <html>\
+    &#9888; <b> Cody encountered an error when processing your message:</b><br>\
```
Comment (mkondratek) [inline src/main/resources/CodyBundle.properties:48]: I wonder... maybe we should put a link to GitHub bug template here? 🤔

File: src/main/kotlin/com/sourcegraph/cody/agent/CodyAgentService.kt [hunk 3]
```diff
@@ -111,33 +113,53 @@ class CodyAgentService(project: Project) : Disposable {
   }
 
   companion object {
+    private val logger = Logger.getInstance(CodyAgent::class.java)
+
+    val agentError: AtomicReference<String?> = AtomicReference(null)
```
Comment (mkondratek) [inline src/main/kotlin/com/sourcegraph/cody/agent/CodyAgentService.kt:105]: nit: `s/agentError/mostRecentlyReportedError/`?

**Review by mkondratek — CHANGES_REQUESTED** [review]

File: src/main/resources/CodyBundle.properties (hunk 2 above)
Comment (mkondratek) [reply src/main/resources/CodyBundle.properties:48]: (users should be able to easily report cases like https://github.com/sourcegraph/jetbrains/issues/487)

File: src/main/kotlin/com/sourcegraph/cody/agent/CodyAgentService.kt (hunk 3 above)
Comment (pkukielka) [reply src/main/kotlin/com/sourcegraph/cody/agent/CodyAgentService.kt:105]: All of those errors are `recently reported` as they are used in the status bar.
So really I do not think adding it only for this element (or at all) adds any information here.

File: src/main/resources/CodyBundle.properties (hunk 2 above)
Comment (pkukielka) [reply src/main/resources/CodyBundle.properties:48]: I'm on the edge there.
Example you gave is good one.
On the other hand things like missing token are not things we would like to be reported.

Maybe @danielmarquespt should look at this and advise, I can do follow-up PR to adjust the message.

File: src/main/kotlin/com/sourcegraph/cody/agent/CodyAgentService.kt (hunk 3 above)
Comment (mkondratek) [reply src/main/kotlin/com/sourcegraph/cody/agent/CodyAgentService.kt:105]: no strong feelings. my suggestion seems to better reflect the variable sense

File: src/main/resources/CodyBundle.properties (hunk 2 above)
Comment (mkondratek) [reply src/main/resources/CodyBundle.properties:48]: We should not leave the user with "Invalid access token." for sure 😄 
For that particular case we probably can suggest the user to re-authenticate. 
For a general case - I'd rather leave a link to bug report.

**Review by mkondratek — APPROVED** [review]
LGTM. let's improve the error handling in next PRs if needed

**Pull request merged**